pkg/updater/updater.go

```diff
@@ -432,3 +353,71 @@ func checkAvailableDiskSpace(fsDisk disk, paths ...string) error {
 	}
 	return nil
 }
+
+type requestKey int
+
+var requestStateKey requestKey
+
+// requestState represents the state of a task.
+type requestState struct {
+	Package string
+	ID      string
+	State   pbgo.TaskState
+	Err     *updaterErrors.UpdaterError
+}
+
+func newRequestContext(request remoteAPIRequest) context.Context {
+	return context.WithValue(context.Background(), requestStateKey, &requestState{
+		Package: request.Package,
+		ID:      request.ID,
+		State:   pbgo.TaskState_RUNNING,
+	})
+}
+
+func setRequestInvalid(ctx context.Context) {
+	state := ctx.Value(requestStateKey).(*requestState)
+	state.State = pbgo.TaskState_INVALID_STATE
+}
+
+func setRequestDone(ctx context.Context, err error) {
+	state := ctx.Value(requestStateKey).(*requestState)
+	state.State = pbgo.TaskState_DONE
+	if err != nil {
+		state.State = pbgo.TaskState_ERROR
+		state.Err = updaterErrors.From(err)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏Not for this PR, but we ought to use the updater errors, I don't think it's done today 😓